src/antsibull_core/dependency_files.py

```diff
@@ -49,14 +50,27 @@ def parse_pieces_file(pieces_file: str) -> list[str]:
     return collections
 
 
-def _parse_name_version_spec_file(filename: str) -> DependencyFileData:
+def parse_pieces_mapping_file(pieces_file: StrPath) -> dict[str, str]:
+    """
+    Parse a pieces file into a Python dictionary
+    """
+
+    lines = parse_pieces_file(pieces_file)
+    mapping: dict[str, str] = {}
+    for line in lines:
+        entries = [entry.strip() for entry in line.split(":", 1)]
+        if len(entries) == 1:
+            raise InvalidFileFormat(f"Invalid line: {line!r}. No ':' found.")
+        mapping[entries[0]] = entries[1]
```

Comment:
```suggestion
        if entries[0] in mapping:
            raise InvalidFileFormat(f"Duplicated key: {entries[0]!r}")
        mapping[entries[0]] = entries[1]
```

Reply:
(I would probably label this a bugfix. It is a change in behavior, but I think an acceptable one.)

Reply:
What do you think about https://github.com/ansible-community/antsibull-core/pull/88/commits/86d58924dfd33c5a7c3daa7272e1dd8eeca0d6ca?